dlc/src/lib.rs

```diff
@@ -0,0 +1,1029 @@
+    accept_fund_pubkey: &PublicKey,
+    output_amount: u64,
+    offer_inputs: &[TxIn],
+    accept_inputs: &[TxIn],
+    offer_change_output: TxOut,
+    accept_change_output: TxOut,
+    lock_time: u32,
+) -> Transaction {
+    let script = make_funding_redeemscript(&offer_fund_pubkey, &accept_fund_pubkey);
+
+    let output: Vec<TxOut> = {
+        let outs = vec![
+            TxOut {
+                value: output_amount,
+                script_pubkey: script.to_v0_p2wsh(),
+            },
+            offer_change_output,
+            accept_change_output,
+        ];
+        outs.into_iter().filter(|o| o.value >= DUST_LIMIT).collect()
+    };
+
+    let input = [&offer_inputs[..], &accept_inputs[..]].concat();
+
+    let funding_transaction = Transaction {
+        version: TX_VERSION,
+        lock_time: lock_time,
+        input,
+        output,
+    };
+
+    return funding_transaction;
+}
+
+/// Create a refund transaction
+pub fn create_refund_transaction(
+    offer_output: TxOut,
+    accept_output: TxOut,
+    funding_input: TxIn,
+    locktime: u32,
+) -> Transaction {
+    Transaction {
+        version: TX_VERSION,
+        lock_time: locktime,
+        input: vec![funding_input],
+        output: vec![offer_output, accept_output],
+    }
+}
+
+/// Create the multisig redeem script for the funding output
+pub fn make_funding_redeemscript(a: &PublicKey, b: &PublicKey) -> Script {
+    let (first, second) = if a <= b { (a, b) } else { (b, a) };
+
+    Builder::new()
+        .push_opcode(opcodes::all::OP_PUSHNUM_2)
+        .push_slice(&first.serialize())
+        .push_slice(&second.serialize())
+        .push_opcode(opcodes::all::OP_PUSHNUM_2)
+        .push_opcode(opcodes::all::OP_CHECKMULTISIG)
+        .into_script()
+}
+
+/// Create an adaptor signature for the given cet
+pub fn create_cet_adaptor_sig_from_point<C: secp256k1::Signing>(
+    secp: &secp256k1::Secp256k1<C>,
+    cet: &Transaction,
+    adaptor_point: &PublicKey,
+    funding_sk: &SecretKey,
+    funding_script_pubkey: &Script,
+    fund_output_value: u64,
+) -> Result<(AdaptorSignature, AdaptorProof), Error> {
+    let sig_hash = util::get_sig_hash_msg(cet, 0, funding_script_pubkey, fund_output_value);
+
+    Ok(secp.adaptor_sign(&sig_hash, &funding_sk, adaptor_point))
+}
+
+/// Create an adaptor signature for the given cet
+pub fn create_cet_adaptor_sig_from_oracle_info<C: secp256k1::Signing>(
+    secp: &secp256k1::Secp256k1<C>,
+    cet: &Transaction,
+    oracle_pubkey: &SchnorrPublicKey,
+    oracle_r_value: &SchnorrPublicKey,
+    funding_sk: &SecretKey,
+    funding_script_pubkey: &Script,
+    fund_output_value: u64,
+    msg: &Message,
+) -> Result<(AdaptorSignature, AdaptorProof), Error> {
+    let adaptor_point = secp.schnorrsig_compute_sig_point(msg, oracle_r_value, &oracle_pubkey)?;
+    let sig_hash = util::get_sig_hash_msg(cet, 0, funding_script_pubkey, fund_output_value);
+
+    Ok(secp.adaptor_sign(&sig_hash, &funding_sk, &adaptor_point))
```

Comment:
nit: might be nicer to have this function call the other one after computing the adaptor point, this way there is less code duplication making changes in the future easier

Reply:
Yeah definitely that was a mistake tbh thanks!